lib/nerves_hub/client.ex

```diff
@@ -0,0 +1,47 @@
+defmodule NervesHub.Client do
+  @moduledoc """
+  Responsible for determining if an update should be applied.
+  """
+  require Logger
+
+  @typedoc "Update that comes over a socket."
+  @type update_data :: map()
+
+  @callback update_available(update_data) :: :apply | :ignore | {:reschedule, pos_integer()}
+
+  @callback reboot_required() :: :apply | :ignore | {:reschedule, pos_integer()}
+
+  def update_available(client, data) do
+    case apply(client, :update_available, [data]) do
```

Comment:
Checking return values here like this feels strange in Elixir. Can this be removed? It seems like the code that calls this function will naturally have a `case` clause that could handle a default operation and also that crashing  on bad return codes wouldn't be unexpected.

Reply:
@fhunleth I chose to do this in case a device has a bad "client" implementation, it would allow the update to be applied. It _can_ be removed if we want though.